### ISSUE TITLE
Ability to edit Pre-set Temp Target

### ISSUE THREAD

**Issue by debbyfurst** [Artificial-Pancreas/iAPS#410]
It would be nice if there was an option to edit a pre-set temp target as opposed to just deleting it and re-creating. That would also allow someone to view the pre-set information in case the user doesn't remember the details. This would prevent someone from making a more drastic change than they intended. For example: If I have a pre-set for sick days, but I don't quite remember what the percent was (because it was created three months earlier when I was sick), I can view the pre-set and then edit it from the existing profile to adjust up or down as needed.

**Comment by dnzxy** [comment]
Hi @debbyfurst , are we talking about temp targets or profiles here? There already is a feature request for the editability of profiles #193 

The "information row" above the chart shows you what percentages and "elements" both a temp target and a profile are currently running on. This does not replace the need for proper editability of presets, but the information is already being shown in *some* regard.

**Comment by debbyfurst** [comment]
Hi Deniz,I’m talking about temp target pre-sets. The information only shows up once you have enacted it. So if you enact a pre-set but then realize it wasn’t the percentage you need, you now have to take extra steps to 1) cancel2) possibly delete the pre-set3) create a new pre-set with the parameters you want (like a different target or percent)If there was the ability to edit or view the pre-set prior to enacting it, that would be more efficient. Thanks,DebbyOn Dec 5, 2023, at 8:57 AM, Deniz Cengiz ***@***.***> wrote:﻿
Hi @debbyfurst , are we talking about temp targets or profiles here? There already is a feature request for the editability of profiles #193
The "information row" above the chart shows you what percentages and "elements" both a temp target and a profile are currently running on. This does not replace the need for proper editability of presets, but the information is already being shown in some regard.

**Comment by dnzxy** [comment]
Yes, you are describing the current limitation of non-editable temp target (as well as profile) presets. Temp targets and profiles are more or less redundant in what they do, but one is lacking some of the integration that the other possesses (under the hood). I'm not sure if going forward both features will be kept, as they more or less to the same. The general editability of presets is a thing currently missing in the app, so thanks for reporting this specifically for temp targets. Noted :)

**Comment by debbyfurst** [comment]
Great. Thanks. On Dec 5, 2023, at 9:38 AM, Deniz Cengiz ***@***.***> wrote:﻿
Yes, you are describing the current limitation of non-editable temp target (as well as profile) presets. Temp targets and profiles are more or less redundant in what they do, but one is lacking some of the integration that the other possesses (under the hood). I'm not sure if going forward both features will be kept, as they more or less to the same. The general editability of presets is a thing currently missing in the app, so thanks for reporting this specifically for temp targets. Noted :)

**Comment by Jon-b-m** [comment]
Edits of profile overrides implemented. Please update your iAPS version.
